File: Lista2 java/src/Pieciokat.java
public class Pieciokat extends Figura{

	private final double bok;

	public Pieciokat(double bok) {
		this.bok = bok;
	}

	@Override
	public double pole() {
		return (Math.sqrt(25+10*Math.sqrt(5))/4) * bok * bok;
	}

	@Override
	public double obwod() {
		return 5 * bok;
	}

	public double getBok() {
		return bok;
	}

	@Override
	public String toString() {
		return "Pieciokat";
	}
}

File: Lista2 cpp/src/Pieciokat.cpp
#include "funs.hpp"
#include <math.h>

Pieciokat::Pieciokat(double a) {
    this->a = a;
}

double Pieciokat::obwod() {
    return 5 * a;
}

double Pieciokat::pole() {
    return sqrt(25+10*sqrt(5)/4) * a * a;
}
